### PR TITLE
Add debug instructions for Cloud Endpoints not being registered.

### PR DESCRIPTION
**Pull request by jlewi**

### CODE REVIEW

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 1]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
 
  * Make sure you are using HTTPS
 
+### DNS name not registered
+
+This section provides troubleshooting information for problems creating a DNS entry for your ingress. This section assumes
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: This is the first time we use the term "ingress" on this page. It'd be good to give more context about what we mean by "ingress".

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 2]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
...
+you are using Cloud Endpoints and the DNS name
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: s/and the DNS name/and a DNS name of the following pattern:

Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: Add a URL for "Cloud Endpoints" to give the reader an idea what we're talking about:
https://cloud.google.com/endpoints/

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 3]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
...
+
+```
+https://${DEPLOYMENT_NAME}.endpoints.${PROJECT}.cloud.goog
+```
+
+Symptoms
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: s/Symptoms/Symptoms:

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 4]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
...
+
+  * When you access the the URL in chrome you get the error: **server IP address could not be found**
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: s/chrome/Chrome

Question: Does the error show up only in Chrome?

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 5]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
...
+
+    ```
+    nslookup ${DEPLOYMENT_NAME}.endpoints.${PROJECT}.cloud.goog
+    Server:   127.0.0.1
+    Address:  127.0.0.1#53
+
+    ** server can't find ${DEPLOYMENT_NAME}.endpoints.${PROJECT}.cloud.goog: NXDOMAIN
+    ```
+
+Troubleshooting
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: s/Troubleshooting/Troubleshooting:

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 6]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
...
+
+1. Check the cloud endpoints resource
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: s/cloud endpoints/Cloud Endpoints

(The general rule is to use capital letters for all brand names, and to avoid capital letters otherwise.)

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 7]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
...
+
+   ```
+   kubectl get cloudendpoints -o yaml ${DEPLOYMENT_NAME}
+   kubectl describe cloudendpoints ${DEPLOYMENT_NAME}
+   ```
+
+   * Check if there are errors indicating problems creating the endpoint
+
+1. The status of the cloudendpoints object will contain the cloud operation used to register the operation
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: Add code tags around `cloudendpoints`

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 8]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
...
+
+   * e.g
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: s/e.g/for example:

A tech writer's tip: Always spell out “i.e.” as “that is”, and “e.g.” as “for example”.
Reason: Very many people don't know the difference between "e.g." and "i.e.". The resulting confusion can be quite bad. Also, abbreviations are hard to parse, particularly for people whose first language is other than English.

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 9]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
...
+        endpoint: jlewi-1218-001.endpoints.cloud-ml-dev.cloud.goog
+        ingressIP: 35.227.243.83
+        jwtAudiences: null
+        lastAppliedSig: 4f3b903a06a683b380bf1aac1deca72792472429
+        observedGeneration: 1
+        stateCurrent: ENDPOINT_SUBMIT_PENDING
+
+     ```
+
+  * You can check the status of the operation
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: It's best to include the colon ":" at the end of this sentence, and all similar sentences in this PR. (It's tempting to assume people will know that the code below is an explanation of how to do the thing, but that's often assuming too much when the reader is working at the edge of their comprehension of this particular topic.)

File: content/docs/guides/gke/troubleshooting-gke.md [hunk 10]
```diff
@@ -24,6 +24,64 @@ Here are some tips for troubleshooting Cloud IAP.
...
+
+    ```
+    gcloud --project=${PROJECT} endpoints operations describe ${OPERATION}
+    ```
+
+    * Operation is everything after "operations/" in the configSubmit field
```
Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: For code readability, add an empty line after this line.

Comment (sarahmaddox) [inline content/docs/guides/gke/troubleshooting-gke.md]: Put code formatting around `configSubmit` and `operations/`.

File: content/docs/guides/gke/troubleshooting-gke.md (hunk 4 above)
Comment (jlewi) [reply content/docs/guides/gke/troubleshooting-gke.md]: No but you might get different error messages in different web browsers.

File: content/docs/guides/gke/troubleshooting-gke.md (hunk 7 above)
Comment (jlewi) [reply content/docs/guides/gke/troubleshooting-gke.md]: Is this what you meant i.e backticks like `cloudendpoints`

**Comment by sarahmaddox** [comment]
Thanks @jlewi! Some bits can still do with a bit of polishing, from a language and readability point of view, but that's true of the page as a whole. Let's get this info out there, then tackle the fine points later.

/lgtm

**Comment by jlewi** [comment]
Thanks
/approve
